pkg/apply/event/event.go

```diff
@@ -72,7 +72,8 @@ type ResourceGroup struct {
 }
 
 type ErrorEvent struct {
-	Err error
+	Object runtime.Object
```

Comment:
This isn't the right place to add this I think. We can discuss the event hierarchy, but I think we want to add errors to the existing event types for apply/prune... The idea is that the top-level error event is for fatal errors, so the client can assume that after receiving one of these error events, the channel will be closed.

Reply:
Ideally we'd like include the error in the `ApplyEvent` or `PruneEvent`. For `PruneEvent`, we can make that change since the objects to be pruned are handled one by one. It's straightforward to add an error in the returned `PruneEvent`. The 'ApplyEvent` is blocked by some limitation in the current code: The `ApplyEvent` is sent out only when the object is applied successfully. The `ApplyEvent` is sent out from the kubectl library, it seems not feasible to inject the error in it.

Reply:
I figured out a way of doing so. When the kubectl `ApplyOptions.Run` fails, we send an ApplyEvent with an error.